app/models/public_body.rb

```diff
@@ -318,10 +323,12 @@ def calculated_home_page
 
   # The "internal admin" is a special body for internal use.
   def self.internal_admin_body
-    # Use find_by_sql to avoid the search being specific to a
-    # locale, since url_name is a translated field:
-    sql = "SELECT * FROM public_bodies WHERE url_name = 'internal_admin_authority'"
-    matching_pbs = PublicBody.find_by_sql sql
+    matching_pbs = PublicBody.
+                     joins(:translations).
+                       where('public_body_translations.locale = ?',
+                             I18n.default_locale).
+                         where(:url_name => 'internal_admin_authority')
```

Comment:
Lets add some tests to make sure this is working as expected. See belgium where they have empty `fr_BE`, `nl_BE` translations and a populated `en` translation (but `en` is not an available locale).